kernel/include/ksched.h

```diff
@@ -526,7 +526,18 @@ static inline struct k_thread *_unpend_first_thread(_wait_q_t *wait_q)
  */
 static inline int _is_thread_user(void)
 {
+#ifdef CONFIG_ARCH_HAS_CUSTOM_SWAP_TO_MAIN
+	/* the _current might be NULL before the first thread is scheduled if
+	 * CONFIG_ARCH_HAS_CUSTOM_SWAP_TO_MAIN is enabled.
+	 */
+	if (!_current) {
```

Comment:
bleah. don't like adding overhead here just for a situation that only happens in the pre-boot environment.
i believe it's the call to k_object_grant_access() in setup_new_thread() which does this.
i'll take this for now, but I'm going to optimize this away later.